analyze/analyze.rs

```diff
@@ -212,48 +213,70 @@ impl traits::Emit for DominatorTree {
             (Align::Left, "Dominator Tree".to_string()),
         ]);
 
+        let dominators = &self.dominators;
+
         fn recursive_add_rows(
             table: &mut Table,
             items: &ir::Items,
             dominator_tree: &BTreeMap<ir::Id, Vec<ir::Id>>,
             depth: usize,
+            dominators: &opt::Dominators,
             id: ir::Id,
         ) {
             assert_eq!(id == items.meta_root(), depth == 0);
 
-            if depth > 0 {
-                let item = &items[id];
+            if dominators.max_row != 0 && depth != dominators.max_depth {
```

Comment:
Rather than indenting the whole function inside this `if`, let's:

* add a new parameter, `row: &mut usize`
* add an early return at the start of this function `if *row == opts.max_rows || depth > opts.max_depth { return; }`
* whenever we actually add a row to the table, `*row += 1;`
* finally, when we first call `recursive_add_rows`, pass `&mut 0` for the `rows` parameter

These tweaks should make this functionality a little less invasive